src/.editorconfig

```diff
@@ -0,0 +1,242 @@
+# editorconfig.org
+# Copied from the .NET Core repo https://github.com/dotnet/corefx/blob/master/.editorconfig
```

Comment:
Doesn't this produce a ton of warnings / errors? We should get them fixed (possibly in a different PR).

Reply:
Yes, I added a few additional rules to remove a bunch of warnings. More can be removed in a separate PR